src/PharoDocComment-Tests/DocCommentsReleaseTest.class.st

```diff
@@ -0,0 +1,25 @@
+"
+This test class builds a test suite from all the doc comments in the system.
+
+See #buildSuite on the class side.
+"
+Class {
+	#name : #DocCommentsReleaseTest,
+	#superclass : #TestCase,
+	#category : #'PharoDocComment-Tests'
+}
+
+{ #category : #'building suites' }
+DocCommentsReleaseTest class >> buildSuite [
+		| suite methods |
+	suite := TestSuite named: 'Test Generated From Comments'.
+	"we do not run examples in tests (as they are used just for testing and might be broken by
+	design"
+	methods := Smalltalk globals allMethods reject: [ :method | method methodClass isTestCase ].
```

Comment:
can you remove the Smalltalk globals ref?

Reply:
yes. But it shows that we are missing a mechanism: the only thing I can do is to use the environment of the test class. But that is not what I mean: I want to write "System, give me all methods, regardless of the environment that this class is in". 

Of course "Smalltalk globals" is not good: for one it is a global reference. And it hardcodes "give me all methods from the global environment", which is not the same as "give me all methods, not only those from classes in the global environment"

Right now there is only one environment, so "self class environment" and "Smalltalk globals" happen to point to the same environment.

I will use self class environment for now.